bootstrap/src/basis/file.ml

```diff
@@ -61,20 +61,29 @@ let stderr =
 external user_data_decref: uns -> unit = "hm_basis_file_user_data_decref"
 external complete_inner: uns -> sint = "hm_basis_file_complete_inner"
 
-let finalize_user_data user_data =
+let register_user_data_finalizer user_data =
   let () = Stdlib.Gc.finalise user_data_decref user_data in
```

Comment:
There's a possibility that `user_data` comes back uninitialized as a zero. Check for that here.